AutoCollection/HttpDependencies.ts

```diff
@@ -74,29 +74,32 @@ class AutoCollectHttpDependencies {
                 }
             }
 
-            (<any>request)[AutoCollectHttpDependencies.alreadyAutoCollectedFlag] = true;
-
             if (request && options && shouldCollect) {
                 CorrelationContextManager.wrapEmitter(request);
-                // If there is no context create one, this apply when no request is triggering the dependency
-                if (!CorrelationContextManager.getCurrentContext()) {
-                    // Create correlation context and wrap execution
-                    let operationId = null;
-                    if (CorrelationIdManager.w3cEnabled) {
-                        let traceparent = new Traceparent();
-                        operationId = traceparent.traceId;
+                if (this._isEnabled) {
+                    // Mark as auto collected
+                    (<any>request)[AutoCollectHttpDependencies.alreadyAutoCollectedFlag] = true;
```

Comment:
Note: for browsers there has been at least 1 environment which doesn't allow "adding" properties to the request, which throws. So you might want to wrap this in a try / catch to allow the telemetry to still be sent. Web also checks the URL